choreolib/src/main/native/include/choreo/trajectory/Trajectory.h

```diff
@@ -0,0 +1,187 @@
+// Copyright (c) Choreo contributors
+
+#pragma once
+
+#include <algorithm>
+#include <optional>
+#include <ranges>
+#include <string>
+#include <vector>
+
+#include <wpi/json_fwd.h>
+
+#include "choreo/trajectory/DifferentialSample.h"
+#include "choreo/trajectory/EventMarker.h"
+#include "choreo/trajectory/SwerveSample.h"
+#include "choreo/trajectory/TrajectorySample.h"
+
+namespace choreo {
+template <TrajectorySample SampleType>
+class Trajectory {
+ public:
+  Trajectory() = default;
+  Trajectory(const std::string& name, const std::vector<SampleType>& samples,
+             const std::vector<int>& splits,
+             const std::vector<EventMarker>& events)
+      : name{name}, samples{samples}, splits{splits}, events{events} {}
+  std::optional<SampleType> GetInitialState() {
```

Comment:
The function spacing is pretty bad in this file. Please include linebreaks between function implementations.